src/components/MovieList.vue

```diff
@@ -0,0 +1,134 @@
+<template>
+  <div>
+    <SortingComponent
+      :movieCount="movieCount"
+      :defaultSortType="defaultSortType"
+      :isDetail="isDetail"
+      :genres="genres"
+      @sortBy="handleSort"
+    />
+    <div v-if="list && list.length" class="movie-list">
+      <MovieCard
+        v-for="movie in list"
+        :movie="movie"
+        :key="movie.title + movie.id"
+        @onDetail="onClickChild"
+      />
+    </div>
+    <div v-else class="movie-list">
+      <span class="movie-list__message"> No movies that match your query </span>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+import SortingComponent from "./SortingComponent.vue";
+import MovieCard from "../shared/MovieCard.vue";
+import { IMovie } from "../interfaces/movie.interface";
+import { ISearch } from "../interfaces/search.interface";
+import useEventBus from "../hooks/useEventBus";
+
+export default defineComponent({
+  name: "MovieList",
+  components: { MovieCard, SortingComponent },
+  props: {
+    movies: Array as () => IMovie[],
+    isDetail: Boolean,
+    genres: Array,
+  },
+  data: function () {
+    return {
+      list: this.movies,
+      defaultSortType: "RATING",
+      movieCount: this.movies?.length,
+      isSearch: false,
+    };
+  },
+  updated: function () {
+    if (!this.isSearch) {
+      this.list = this.movies;
+    } else {
+      this.isSearch = false;
```

Comment:
hmm... I would use watch here as we are interested only in isSearch property and not every update https://v2.vuejs.org/v2/api/#vm-watch